scripts/data/common/killed-in-gaza/reconcile_lists.ts

```diff
@@ -128,20 +131,27 @@ const validateDobAgeWithinYear = (
   const diff = Math.abs(age - ageFromDob);
   return [diff < 2, flipped ? dob : undefined];
 };
+
+// if dob exists, calculate age from dob and reference date
+// else use existing age value from the record
 const addExistingAge = (record: ExistingRecord) => {
   const dobDate = record.dob ? new Date(record.dob) : null;
-  if (dobDate && Number.isNaN(dobDate.getTime())) {
-    throw new Error(
-      `Invalid date found in addExistingAge transform: ${record.dob}`
-    );
+  let age: string | undefined
+  if(dobDate) {
```

Comment:
i think you may want to initialize with the existing record's age so that we're not rewriting the DOB from a ministry-provided one to one that's "estimated" using the january reference date

```suggestion
  let age = record.age
  if(dobDate && !age) {
```

you'll probably want to also update the type at the top of the file since existing records do have an age now since the last update:

```diff
type ExistingRecord = {
  id: string;
  name_ar_raw: string;
  dob: string;
  sex: "M" | "F";
+ age: string
};
```

you could then update the return to just be

```
return { ...record, age }
```

great catch here though! forgot that this was here to add the age since we didn't track it in raw.csv before

Reply:
on second thought - i think we can remove that function altogether now that both datasets have an age

something like this maybe:
https://github.com/TechForPalestine/palestine-datasets/compare/20240525_data_rm_age_fn